### ISSUE TITLE
var-naming[no-role-prefix] false positive with molecule roles

### ISSUE THREAD

**Issue by alessfg** [ansible/ansible-lint#3683]
##### Summary

Ansible Lint detects the following code block as violating the `var-naming[no-role-prefix]` rule. This is specially an issue when testing roles using Molecule, since the role name might not match the name used in the Molecule converge file (see below example). 

This issue happens both when specifying a variable as the role name, or when using the project/repository name if the project name does not match the actual role name (the project for this role is called `ansible-role-nginx`, while the actual role name is `nginxinc.nginx`, and in order for Molecule to properly work in local environments I need to use the project name instead of the role name).

```yaml
---
- name: Converge
  hosts: all
  tasks:
    - name: Install NGINX
      ansible.builtin.include_role:
        name: "{{ lookup('env', 'MOLECULE_PROJECT_DIRECTORY') | basename }}"
        # name: ansible-role-nginx  #### Also fails
      vars:
        nginx_logrotate_conf_enable: true
        nginx_logrotate_conf:
          paths:
            - /var/log/nginx/*.log
          options:
            - daily
```

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
$ ansible-lint --version
ansible-lint 6.18.0 using ansible-core:2.15.3 ansible-compat:4.1.7 ruamel-yaml:0.17.25 ruamel-yaml-clib:0.2.7
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

Run Ansible Lint on the following playbook:

```yaml
---
- name: Converge
  hosts: all
  tasks:
    - name: Install NGINX
      ansible.builtin.include_role:
        name: "{{ lookup('env', 'MOLECULE_PROJECT_DIRECTORY') | basename }}"
        # name: ansible-role-nginx  #### Also fails
      vars:
        nginx_logrotate_conf_enable: true
        nginx_logrotate_conf:
          paths:
            - /var/log/nginx/*.log
          options:
            - daily
```

##### Desired Behavior

Ansible Lint should not run this rule:
* Inside Molecule converge playbooks.
* When a variable has been set as the role name.

##### Actual Behavior

```console
$ ansible-lint
WARNING  Listing 5 violation(s) that are fatal
var-naming[no-role-prefix]: Variables names from within roles should use role_name_ as a prefix. (vars: nginx_logrotate_conf)
molecule/default/converge.yml:9 Task/Handler: Install NGINX

var-naming[no-role-prefix]: Variables names from within roles should use role_name_ as a prefix. (vars: nginx_logrotate_conf_enable)
molecule/default/converge.yml:9 Task/Handler: Install NGINX
```
